src/main/generic/api/Client.js

```diff
@@ -762,7 +762,7 @@ class Client {
         for (const receipt of receipts.valueIterator()) {
             // Skip known transactions that are already considered confirmed.
             const knownTx = knownTxs.get(receipt.transactionHash);
-            if (knownTx && knownTx.blockHash.equals(receipt.blockHash) && knownTx.state === Client.TransactionState.CONFIRMED) {
+            if (knownTx && knownTx.state === Client.TransactionState.CONFIRMED && knownTx.blockHash.equals(receipt.blockHash)) {
```

Comment:
Can you change to do `receipt.blockHash.equals(knownTx.blockHash) `? Because receipt is guaranteed to have a blockHash by it's constructor and for knownTx it's optional.

Reply:
Done :+1: